dalle2_pytorch/trackers.py

```diff
@@ -398,20 +419,65 @@ def __init__(self, data_path: Optional[str] = DEFAULT_DATA_PATH, overwrite_data_
         self.savers: List[BaseSaver]= []
         self.dummy_mode = dummy_mode
 
+    def _load_auto_resume(self) -> bool:
+        # First, we check if data_path / auto_resume.json exists
+        if self.auto_resume_path.exists():
+            # First, we check if the logger has autoresume enabled. If not, we remove the autoresume file and move on
+            if not self.logger.auto_resume:
+                print(f'Removing auto_resume.json because auto_resume is not enabled in the config')
+                self.auto_resume_path.unlink()
+                return False
+            else:
+                # Otherwise we read the json into a dictionary will will override parts of logger.__dict__
+                with open(self.auto_resume_path, 'r') as f:
+                    auto_resume_dict = json.load(f)
+                # Check if the logger is of the same type as the autoresume save
+                if auto_resume_dict['logger_type'] != self.logger.__class__.__name__:
+                    raise Exception(f'The logger type in the auto_resume file is {auto_resume_dict["logger_type"]} but the current logger is {self.logger.__class__.__name__}. Either use the original logger type, set `auto_resume` to `False`, or delete your existing tracker-data folder.')
+                # Then we are ready to override the logger with the autoresume save
+                self.logger.__dict__["resume"] = True
+                print(f"Upading {self.logger.__dict__} with {auto_resume_dict}")
+                self.logger.__dict__.update(auto_resume_dict)
+
+                return True
+        else:
```

Comment:
more a stylistic critique, but one way for a cleaner function when you have all these branching nested if / else, is to put the edge cases near the top and do an early return

so in your function here

```python
if not self.auto_resume_path.exists():
  # log
  return False

if not self.logger.auto_resume:
  # log
  return False

# rest of the main logic loading from json + logging
return True
```

that way it is more readable and the edge cases are organized near the top